### DISCUSSION TITLE
How to access GuildId with global commands?

### DISCUSSION THREAD

**Discussion by Halldor-Hrafn** [serenity-rs/serenity#2447] [Q&A] [answered]
I am fairly new to Serenity, and I am wondering about something.

When I register a slash command as a guild command, I always get a `GuildId` sent with the `CommandData`.
But when I register the same command as a global command, I get a `None` instead of a GuildId.

Long story short, how would I get the `GuildId` with a global command?

Oh, also, is it possible to get `UserId` aswell? just curious as the bot I'm making would need both for what I want it to do.

**Comment by arqunis** [comment, marked as answer]
> Long story short, how would I get the `GuildId` with a global command?

`CommandData::guild_id` represents which guild the command was *registered* to. It is not the guild where the interaction was used. [You should inspect the interaction's data for that instead](https://serenity-rs.github.io/serenity/current/serenity/model/application/interaction/application_command/struct.ApplicationCommandInteraction.html#structfield.guild_id).

[The same applies for the user's ID](https://serenity-rs.github.io/serenity/current/serenity/model/application/interaction/application_command/struct.ApplicationCommandInteraction.html#structfield.user).

**Reply by Halldor-Hrafn** [reply]
I was so fixated on looking at the `CommandData` struct that I didn't even think of looking at the `ApplicationCommandInteraction` struct.

This makes so much more sense than what I was thinking of.

Thanks for the help.
